subsys/net/lib/sockets/sockets.c

```diff
@@ -202,12 +271,29 @@ int zsock_listen(int sock, int backlog)
 	return 0;
 }
 
-int zsock_accept(int sock, struct sockaddr *addr, socklen_t *addrlen)
+#ifdef CONFIG_USERSPACE
+Z_SYSCALL_HANDLER(zsock_listen, sock, backlog)
+{
+	if (Z_SYSCALL_OBJ(sock, K_OBJ_NET_CONTEXT)) {
+		errno = EBADF;
+		return -1;
+	}
+
+	return _impl_zsock_listen(sock, backlog);
+}
+#endif /* CONFIG_USERSPACE */
+
+int _impl_zsock_accept(int sock, struct sockaddr *addr, socklen_t *addrlen)
 {
 	struct net_context *parent = INT_TO_POINTER(sock);
 
 	struct net_context *ctx = k_fifo_get(&parent->accept_q, K_FOREVER);
 
+#ifdef CONFIG_USERSPACE
+	_k_object_init(ctx);
```

Comment:
Well, ctx is gotten from fifo, but it gets into fifo from the native stack, which net_context_get's it. So, I don't see why it wouldn't be _k_object_recycle() here, just the same as for socket(). If you see, that should go into code comments, because many will be confused going forward.